game.js

```diff
@@ -0,0 +1,81 @@
+const Frame = require('./frame');
+const ScoreCard = require('./scoreCard');
+const readline = require('readline');
+
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout
+});
+
+class Game {
+  constructor() {
+    this.scoreCard = new ScoreCard;
+    this.frameCount = 0;
+  }
+
+  startGame() {
+    this.newFrame();
+  }
+
+  newFrame() {
+    this.frameCount += 1;
+    this.frame = new Frame;
+    this.runFrame();
+  }
+
+  runFrame() {
+    this.pinsRequest(this.frame.startRoll()); // needs a callback?
+    if (this.frame.bonus === 'strike') {
+      this.frame.frameScores.push(0);
+    } else {
+      this.pinsRequest(this.frame.startRoll()); // needs a callback
+    }
+    this.finishFrame();
+  }
+
+  finishFrame() {
+    this.frame.frameScores.push(this.frame.bonus);
+    this.scoreCard.frameResult(this.frame.frameScores);
+    this.scoreCard.printScores();
+    this.frameCount < 10 ? this.newFrame() : this.bonus(this.frame.bonus, this.frameCount);
+  }
+
+  finishGame() {
+    if (this.scoreCard.totalScore === 0) {
+      console.log("GUTTER GAME");
+    } else if (this.scoreCard.totalScore === 300) {
+      console.log("PERFECT GAME");
+    }
+    console.log(this.scoreCard.printScores());
+  }
+
+  pinsRequest() {
+    return 10
+  }
+
+  // pinsRequest(callback) {
+  //   rl.question('How many pins did you knock down?', (pins) => {
+  //     console.log(`pins: ${pins}`);
+  //     callback(pins);
+  //   });
+  // }
+
+  bonus(bonusType, frameRound) {
```

Comment:
such a clear way of calculating bonus points - I'm in awe! (Mine is a complete mess!)